### PR TITLE
Added new actions, corrected linter errors in sources #1510

### PR DESCRIPTION
**Pull request by feyzullah**
#1510 
Added actions:
*create-activity
*get-activity-by-id
*get-stats
*list-activities
*update-activity
Linter errors has been fixed in sources

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/strava/actions/create-activity/create-activity.mjs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    hide_from_home: {
+      type: "boolean",
+      label: "Hide from home",
+      description: "Set to true to mute activity",
+      optional: true,
+    },
+  },
+  async run({ $ }) {
+    const isoDateRegexp = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})(([+-](\d{2}):(\d{2})|Z))$/;
```
Comment (vellames-turing) [inline components/strava/actions/create-activity/create-activity.mjs:63]: Could you please add a comment explaining this regexp? just to make it the clear as possible for other people that would eventually maintain this code

File: components/strava/actions/create-activity/create-activity.mjs [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import strava from "../../strava.app.js";
+
+export default {
+  name: "Create Activity",
+  description: "Creates a manual activity for an athlete. See `https://developers.strava.com/docs/reference/`",
```
Comment (vellames-turing) [inline components/strava/actions/create-activity/create-activity.mjs:5]: Could you please use `[]()` to ref the docs?

File: components/strava/actions/get-activity-by-id/get-activity-by-id.mjs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import strava from "../../strava.app.js";
+
+export default {
+  name: "Get Activity By ID",
+  description: "Returns the given activity that is owned by the authenticated athlete. See `https://developers.strava.com/docs/reference/`",
```
Comment (vellames-turing) [inline components/strava/actions/get-activity-by-id/get-activity-by-id.mjs:5]: Could you please use []() to ref the docs?

File: components/strava/actions/get-stats/get-stats.mjs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import strava from "../../strava.app.js";
+
+export default {
+  name: "Get Stats",
+  description: "Returns the activity stats of an athlete. Only includes data from activities set to Everyone visibilty. See `https://developers.strava.com/docs/reference/`",
```
Comment (vellames-turing) [inline components/strava/actions/get-stats/get-stats.mjs:5]: Could you please use []() to ref the docs?

File: components/strava/actions/list-activities/list-activities.mjs [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import strava from "../../strava.app.js";
+
+export default {
+  name: "Get Activity List",
+  description: "Returns the activities of an athlete for a specific identifier. See `https://developers.strava.com/docs/reference/`",
```
Comment (vellames-turing) [inline components/strava/actions/list-activities/list-activities.mjs:5]: Could you please use []() to ref the docs?

File: components/strava/actions/update-activity/update-activity.mjs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import strava from "../../strava.app.js";
+
+export default {
+  name: "Update Activity",
+  description: "Updates the given activity that is owned by the authenticated athlete. See `https://developers.strava.com/docs/reference/`",
```
Comment (vellames-turing) [inline components/strava/actions/update-activity/update-activity.mjs:5]: Could you please use []() to ref the docs?

File: components/strava/strava.app.js [hunk 7]
```diff
@@ -1,21 +1,29 @@
 // Strava API app file
 const axios = require("axios");
```
Comment (vellames-turing) [inline components/strava/strava.app.js:2]: It would be great if you could remove this usage of `axios` and use only `@pipedream/platform`

**Review by vellames-turing — CHANGES_REQUESTED** [review]
Could you please refactor all sources and app to ESModules, please? This way you can import `strava.app.mjs`

File: components/strava/actions/get-stats/get-stats.mjs [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  key: "strava-get-stats",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    strava,
+  },
+  async run({ $ }) {
+    const resp = await this.strava.getStats($);
+    $.export("$summary", "The activity stats have been retrieved");
+    return resp;
```
Comment (jcortes) [inline components/strava/actions/get-stats/get-stats.mjs:15]: Once you make the refactor in `strava.app.js` you can apply these changes
```suggestion
    const { id: athleteId } = await this.strava.getAuthenticatedAthlete({
      $,
    });
    const resp = await this.strava.getAthleteStats({
      $,
      athleteId
    });
    $.export("$summary", "The activity stats have been retrieved");
    return resp;
```

File: components/strava/strava.app.js [hunk 9]
```diff
@@ -1,21 +1,29 @@
...
-
+const { axios: PlatformAxios } = require("@pipedream/platform");
 module.exports = {
```
Comment (jcortes) [inline components/strava/strava.app.js:4]: I would also recommend the migration from CommonJS to ESM modules and renaming all files from `.js` to `.mjs` extension
```suggestion
export default {
```

**Review by jcortes — CHANGES_REQUESTED** [review]
It's looking good overall. I requested a few refactors other than that it looks good!

File: components/strava/actions/create-activity/create-activity.mjs (hunk 1 above)
Comment (feyzullah) [reply components/strava/actions/create-activity/create-activity.mjs:63]: done

File: components/strava/actions/create-activity/create-activity.mjs (hunk 2 above)
Comment (feyzullah) [reply components/strava/actions/create-activity/create-activity.mjs:5]: done

File: components/strava/actions/create-activity/create-activity.mjs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+      type: this.type,
+      start_date_local: this.start_date_local,
+      elapsed_time: this.elapsed_time,
+      description: this.description,
+      distance: this.distance,
+      trainer: this.trainer,
+      commute: this.commute,
+      hide_from_home: this.hide_from_home,
+    };
+    if (this.distance && isNaN(parseFloat(this.distance)))
```
[the remark this answers is not in the document]
Comment (feyzullah) [reply components/strava/actions/create-activity/create-activity.mjs:75]: done

File: components/strava/actions/get-activity-by-id/get-activity-by-id.mjs (hunk 3 above)
Comment (feyzullah) [reply components/strava/actions/get-activity-by-id/get-activity-by-id.mjs:5]: done

File: components/strava/actions/get-stats/get-stats.mjs (hunk 4 above)
Comment (feyzullah) [reply components/strava/actions/get-stats/get-stats.mjs:5]: done

File: components/strava/actions/update-activity/update-activity.mjs (hunk 6 above)
Comment (feyzullah) [reply components/strava/actions/update-activity/update-activity.mjs:5]: done

File: components/strava/strava.app.js (hunk 7 above)
Comment (feyzullah) [reply components/strava/strava.app.js:2]: done

File: components/strava/strava.app.js [hunk 11]
```diff
@@ -31,5 +39,91 @@ module.exports = {
...
+      } = data;
+      return this._listActivitiesOfPage(
+        $,
+        dataRest,
+        maxItems,
+        pageSize,
+        [],
+        1,
+      );
+    },
```
[the remark this answers is not in the document]
Comment (feyzullah) [reply components/strava/strava.app.js:109]: done with small changes

File: components/strava/actions/get-stats/get-stats.mjs (hunk 8 above)
Comment (feyzullah) [reply components/strava/actions/get-stats/get-stats.mjs:15]: done with small changes

File: components/strava/strava.app.js [hunk 12]
```diff
@@ -1,21 +1,29 @@
...
       }${path}`;
-      return await axios(opts);
+      return opts;
+    },
+    async _makeAPIRequest(opts) {
+      return await axios(this.prepareOpts(opts));
+    },
+    async _makeAPIRequestWithPlatform($, opts) {
+      return await PlatformAxios($, this.prepareOpts(opts));
     },
```
[the remark this answers is not in the document]
Comment (feyzullah) [reply components/strava/strava.app.js:27]: done

File: components/strava/strava.app.js [hunk 13]
```diff
@@ -1,21 +1,29 @@
 // Strava API app file
 const axios = require("axios");
-
+const { axios: PlatformAxios } = require("@pipedream/platform");
```
[the remark this answers is not in the document]
Comment (feyzullah) [reply components/strava/strava.app.js:3]: done

File: components/strava/strava.app.js (hunk 9 above)
Comment (feyzullah) [reply components/strava/strava.app.js:4]: done

File: components/strava/actions/list-activities/list-activities.mjs (hunk 5 above)
Comment (feyzullah) [reply components/strava/actions/list-activities/list-activities.mjs:5]: done

**Comment by feyzullah** [comment]
applied suggestions, closing this, opening a new PR

**Pull request closed without merging**